src/box2d.cpp

```diff
@@ -0,0 +1,69 @@
+#include "box2d.hpp"
+
+Box2D::Box2D(Box2D && obj)
+{
+  swap(obj);
+  std::cout << "Move copy constructor" <<std::endl;
+}
+
+Box2D & Box2D::operator = (Box2D const & obj)
+{
+  if (this == &obj) return *this;
+  Box2D tmp(obj);
+  swap(tmp);
+  return *this;
+}
+
+Box2D & Box2D::operator = (Box2D && obj)
+{
+  swap(obj);
+  std::cout << "Move operator=" <<std::endl;
+  return *this;
+}
+
+bool Box2D::operator == (Box2D const & obj) const
+{
+  return m_point1 == obj.m_point1
+      && m_point2 == obj.m_point2;
+}
+
+void Box2D::swap(Box2D & obj)
+{
+  std::swap(m_point1, obj.m_point1);
+  std::swap(m_point2, obj.m_point2);
+}
+
+bool Box2D::checkInside(const Box2D &box, const Point2D &point)
+{
+  if ((point.x() >= box.m_point1.x() &&
```

Comment:
What will be in the following case:
box.boxMin().x() > box.boxMax().x() or box.boxMin().y() > box.boxMax().y()

We don't set any assumption for the box.boxMax() and box.boxMin() values, it could be any.